app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -330,6 +332,7 @@ class ExplorationActivityPresenter @Inject constructor(
       showUnsavedExplorationDialogFragment()
       return
     }
+    learnerAnalyticsLogger.explorationAnalyticsLogger.value?.logLessonSavedAdvertently()
```

Comment:
> The current placement of this call will cause the log to be created every time, even when checkpointing is disabled, as it is outside the if block.

The current placement of this call will not log the event when checkpointing is disabled because the if block just above it checks whether checkpointing is enabled or not. Incase checpointing is disabled the return statement in the if block will be executed and following lines including the log statement will not be executed.

But, you are right, this should be inside `showDialogFragmentBasedOnCurrentCheckpointState()` inside the `when` block to ensure that the event is logged only when lesson checkpoint is saved successfully.